### PR TITLE
fix(evm): validate EIP-7685 requests root against the header

### PR DESCRIPTION
**Pull request by shekhirin**
Ref https://github.com/ethereum/EIPs/pull/8592

Closes https://github.com/paradigmxyz/reth/issues/8301

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by shekhirin** [comment]
Testing on kurtosis rn

**Review by mattsse — APPROVED** [review]

File: crates/consensus/consensus/src/lib.rs [hunk 1]
```diff
@@ -10,14 +10,24 @@
...
 /// test helpers for mocking consensus
 pub mod test_utils;
 
+/// Post execution input passed to [Consensus::validate_block_post_execution].
+#[derive(Debug)]
+pub struct PostExecutionInput<'a> {
+    /// Receipts of the block.
+    pub receipts: &'a [Receipt],
+    /// EIP-7685 requests of the block.
+    pub requests: &'a [Request],
```
Comment (mattsse) [inline crates/consensus/consensus/src/lib.rs:28]: I wonder if we should do Option here?

File: crates/consensus/consensus/src/lib.rs [hunk 2]
```diff
@@ -10,14 +10,24 @@
...
 };
 use std::fmt::Debug;
 
 #[cfg(any(test, feature = "test-utils"))]
 /// test helpers for mocking consensus
 pub mod test_utils;
 
+/// Post execution input passed to [Consensus::validate_block_post_execution].
+#[derive(Debug)]
+pub struct PostExecutionInput<'a> {
```
Comment (mattsse) [inline crates/consensus/consensus/src/lib.rs:24]: let's add a helper function like 

`fn new`
`fn receipts(receipts)`

**Review by mattsse — APPROVED** [review]
lgtm,
one suggestion

File: crates/consensus/consensus/src/lib.rs (hunk 1 above)
Comment (shekhirin) [reply crates/consensus/consensus/src/lib.rs:28]: we return an empty vector from block execution both when hardfork is enabled and when no requests are found. So I think it's easier to leave it as a vector here

**Pull request merged**